engine/grid.py

```diff
@@ -119,12 +286,17 @@ def get_spiral_waypoints(self):
         col = 0  # start at bottom left corner
         row = 0
 
-        step_col = (1, 0, -1, 0)  # these tuples simulate the robot's next movement based on turn state
+        # these tuples simulate the robot's next movement based on turn state
+        step_col = (1, 0, -1, 0)
         step_row = (0, 1, 0, -1)
         turn_state = 0  # turn_state is a variable that must be between 0..3
 
         for _ in range(rows * cols):  # for loop over all nodes
             node = node_list[row, col]
+            if node.is_active:
+                self.active_nodes_list.append(node)
+            else:
+                self.inactive_nodes_list.append(node)
```

Comment:
i think you could abstract this into a function since it is used in many diff functions to simplify code + in case we want to change the way active/inactive is implemented in the future
for ex `def handleNodeActivity(node)` or something with a better name lol

This helper function can go under METHODS TO ACTIVATE NODES ON THE GRID

Reply:
@Shreya1233